src/core/border_router/routing_manager.cpp

```diff
@@ -743,18 +745,8 @@ void RoutingManager::EvaluateRoutingPolicy(void)
         StartRoutingPolicyEvaluationDelay(Time::SecToMsec(nextSendDelay));
     }
 
-    // 3. Update advertised on-link & OMR prefixes information.
-    {
-        bool wasAdvertisingLocalOnLinkPrefix = mIsAdvertisingLocalOnLinkPrefix;
-
-        mIsAdvertisingLocalOnLinkPrefix = (newOnLinkPrefix == &mLocalOnLinkPrefix);
-        if (!wasAdvertisingLocalOnLinkPrefix && mIsAdvertisingLocalOnLinkPrefix)
-        {
-            InvalidateDiscoveredPrefixes();
-        }
```

Comment:
In new push, I moved this logic to `EvaluateOnLinkPrefix()` where we set `mIsAdvertisingLocalOnLinkPrefix` to `true`. Added a comment on why we call this as well. PTAL